dashboard/src/components/AppView/SecretItem.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from "react";
+import { Eye, EyeOff } from "react-feather";
+
+import { ISecret } from "../../shared/types";
+import "./SecretContent.css";
+
+interface ISecretItemProps {
+  secret: ISecret;
+}
+
+interface ISecretItemState {
+  display: { [s: string]: boolean };
+}
+
+class SecretItem extends React.Component<ISecretItemProps, ISecretItemState> {
+  public state: ISecretItemState = {
+    display: {},
+  };
+
+  public componentDidMount() {
+    const display = {};
+    Object.keys(this.props.secret.data).forEach(k => (display[k] = false));
+    this.setState(display);
```

Comment:
I do not think that you need componentDidMount here, I believe that you can use the constructor or the state class propery.

```
  public state: ISecretItemState = {
    display: {},
  };
```

I believe that you can initialize there.